design-documents/graph-ql/coverage/reward-points.graphqls

```diff
@@ -0,0 +1,70 @@
+type Customer {
+    reward_points: RewardPoints @doc(description: "Customer reward points details")
+}
+
+type Cart {
+    applied_reward_points: RewardPointsAmount @doc(description: "Reward points ammount applied to the cart")
+}
+
+type Mutation {
+    applyRewardPointsToCart(cartId: ID!): ApplyRewardPointsToCartOutput @doc(description: "Apply all available points up to the cart total. Partial redemption is not available")
+    removeRewardPointsFromCart(cartId: ID!): RemoveRewardPointsFromCartOutput @doc(description: "Cancel usage of the previously applied reward points to cart")
+}
+
+type ApplyRewardPointsToCartOutput {
+    cart: Cart
+}
+
+type RemoveRewardPointsFromCartOutput {
+    cart: Cart
+}
+
+type RewardPoints {
+    balance: RewardPointsAmount @doc(description: "Current reward points balance")
+    exchange_rates: RewardPointsExchangeRates @doc(description: "Current reward points exchange rates")
+    subscription_status: RewardPointsSubscriptionStatus @doc(description: "Reward points related email subscription status")
+    balance_history: [RewardPointsBalanceHistoryItem] @doc(description: "Reward points balance history. This field will be set to null if disabled in the admin")
+}
+
+type RewardPointsAmount {
+    points: Float! @doc(description: "Reward points amount in points")
+    money: Money! @doc(description: "Reward points amount in store currency")
+}
+
+type RewardPointsExchangeRates @doc (description: "Exchange rates depend on the customer group"){
+    earning: Float! @doc(description: "How many points are earned per 1 currency spent")
+    redemption: Float! @doc(description: "How many points need to be redeemed to get 1 currency discount at the checkout")
+}
+
+type RewardPointsSubscriptionStatus {
+    balance_updates: SubscriptionStatus! @doc(description: "Customer subscription status to 'Reward points balance updates' emails")
+    points_expiration_notifications: SubscriptionStatus! @doc(description: "Customer subscription status to 'Reward points expiration notifications' emails")
+}
+
+enum SubscriptionStatus {
+    SUBSCRIBED
+    NOT_SUBSCRIBED
+}
+
+type RewardPointsBalanceHistoryItem {
+    balance: RewardPointsAmount @doc(description: "Reward points balance after the completion of the transaction")
+    points_change: Float! @doc(description: "Number of points added or deducted in the transaction")
+    change_reason: String! @doc(description: "Reason for balance change")
+    date: String! @doc(description: "Transaction date")
+}
+
+type StoreConfig {
+    magento_reward_general_is_enabled: String @doc(description: "Reward points functionality status: enabled/disabled")
```

Comment:
Is there something in core preventing us from using a Boolean here?

Reply:
The logic here is, we take data from `core_config_data` as is and run it through the list of fields declared in this GraphQL schema to leave only allowed ones. 
These values are currently stored in DB as 0/1. I don't think GraphQL framework will properly convert them to boolean. It may be possible, but will require additional effort to add support.

Reply:
I can definitely see how this makes the implementation easier/lower effort.

My concern is that the work doesn't go away, we're just pushing it to every client instead. Every client now has to do:

```js
if (magento_reward_general_is_enabled === 'enabled') {

}
```

rather than us returning a more descriptive type (`Boolean`)

Reply:
My 2 cents: there is already a lot of unnecessary front-end logic in the client. e.g. mapping and transforming data coming from GraphQL. Although this might seem small, it does make a more descriptive and elegant API. We gotta start trimming logic where we can in order to enable clients to be leaner.

Reply:
I would address it in scope of separate story related to `StoreConfig`. At the time it was implemented we agreed to the tradeoff. 
Otherwise Reward implementation will be blocked by the rework of `StoreConfig`. 

@nrkapoor are you okay with creating a separate story for improving DX of `StoreConfig` query?

Reply:
@paliarush Yes I will create a separate ticket for storeconfig clean.

Reply:
Sounds good - thanks y'all 👍